pipeline/setup-runner.py

```diff
@@ -0,0 +1,319 @@
+
+        print(f"Installing Blender {version}...")
+        blender_dmg = Path(f"/tmp/blender-{version}.dmg")
+
+        if not USE_PUBLIC_URLS and download_from_s3(
+            f"blender/blender-{version}-macos-arm64.dmg", blender_dmg
+        ):
+            pass
+        elif USE_PUBLIC_URLS:
+            run(
+                f"curl -L -o {blender_dmg} https://download.blender.org/release/Blender{major_minor}/blender-{version}-macos-arm64.dmg",
+                shell=True,
+            )
+            verify_checksum(blender_dmg, BLENDER_CHECKSUMS[f"{version}-macos-arm64"])
+
+        run(f"hdiutil attach {blender_dmg}", shell=True)
+        run(f"sudo rm -rf {blender_app}", shell=True, check=False)
+        run(f"sudo cp -R /Volumes/Blender/Blender.app {blender_app}", shell=True)
+        run("hdiutil detach /Volumes/Blender", shell=True)
+
+        blender_dmg.unlink(missing_ok=True)
+
+    print("Installing Blender submitter...")
+    run("hatch build", shell=True)
+
+    submitter_path = Path("./DeadlineCloudSubmitter")
+    if submitter_path.exists():
+        run(f"rm -rf {submitter_path}", shell=True, check=False)
+
+    Path("./DeadlineCloudSubmitter/Submitters/Blender/python").mkdir(parents=True, exist_ok=True)
+    run(
+        "cp -r src/deadline/blender_submitter/addons/ ./DeadlineCloudSubmitter/Submitters/Blender/python/addons",
+        shell=True,
+    )
+
+    run(
+        'pip install --upgrade --python-version 3.11 --only-binary=:all: "deadline[gui]" blender-qt-stylesheet -t ./DeadlineCloudSubmitter/Submitters/Blender/python/modules',
+        shell=True,
+    )
+
+    for version in BLENDER_VERSIONS:
+        blender_exe = f"/Applications/Blender-{version}.app/Contents/MacOS/Blender"
+        run(
+            f"{blender_exe} --background --python ./installer/add_submitter_to_pref.py -- --deadline_cloud_install_path $(pwd)/DeadlineCloudSubmitter/Submitters/Blender/python",
+            shell=True,
+        )
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Setup Blender test environment")
+    parser.add_argument(
+        "--public-urls", action="store_true", help="Download from public URLs instead of S3"
+    )
+    parser.add_argument(
+        "--versions", nargs="+", help="Blender versions to install (e.g., 4.5.4 4.2.12)"
+    )
+    parser.add_argument(
+        "--install-x11",
+        action="store_true",
+        help="Install X11 libraries and start Xvfb (Linux only)",
+    )
+    args = parser.parse_args()
+
+    USE_PUBLIC_URLS = args.public_urls
+    if args.versions:
+        BLENDER_VERSIONS = args.versions
+    system = platform.system()
+    print(f"Setting up {system} with Blender {', '.join(BLENDER_VERSIONS)}")
+    print(f"Using {'public URLs' if USE_PUBLIC_URLS else 'S3 bucket'}")
+    if system == "Linux":
+        setup_linux(install_x11=args.install_x11)
+    else:
+        {"Windows": setup_windows, "Darwin": setup_macos}[system]()
```

Comment:
I wonder if this could be made more simple for maintainability:

```suggestion
    elif system == "Windows":
        setup_windows()
    elif system == "Darwin":
        setup_macos()
    else:
        raise OSError(f"Unsupported OS: {system}")
```

Reply:
Simplified for maintainability.